Snippets/Snippets_6/UpgradeGuides/5to6/CustomErrorHandlingBehaviour.cs

```
@@ -0,0 +1,59 @@
+namespace Snippets6.UpgradeGuides._5to6
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus;
+    using NServiceBus.Pipeline;
+
+    #region 5to6-customerrorhandlingbehaviour
+    class CustomErrorHandlingBehaviour : Behavior<ITransportReceiveContext>
+    {
+        public override async Task Invoke(ITransportReceiveContext context, Func<Task> next)
+        {
+            try
+            {
+                await next().ConfigureAwait(false);
+            }
+            catch (MessageDeserializationException)
```

Why does the snippet code swallow this exception?